src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MatchXpathCheck.java

```diff
@@ -0,0 +1,285 @@
+ * application</a> with options to show AST and to ease usage of Xpath queries.
+ * </p>
+ * <p><b>-T</b> option prints AST tree of the checked file.</p>
+ * <pre>
+ * $ java -jar checkstyle-X.XX-all.jar -T Main.java
+ * CLASS_DEF -&gt; CLASS_DEF [1:0]
+ * |--MODIFIERS -&gt; MODIFIERS [1:0]
+ * |   `--LITERAL_PUBLIC -&gt; public [1:0]
+ * |--LITERAL_CLASS -&gt; class [1:7]
+ * |--IDENT -&gt; Main [1:13]
+ * `--OBJBLOCK -&gt; OBJBLOCK [1:18]
+ * |--LCURLY -&gt; { [1:18]
+ * |--METHOD_DEF -&gt; METHOD_DEF [2:4]
+ * |   |--MODIFIERS -&gt; MODIFIERS [2:4]
+ * |   |   `--LITERAL_PUBLIC -&gt; public [2:4]
+ * |   |--TYPE -&gt; TYPE [2:11]
+ * |   |   `--IDENT -&gt; String [2:11]
+ * |   |--IDENT -&gt; sayHello [2:18]
+ * |   |--LPAREN -&gt; ( [2:26]
+ * |   |--PARAMETERS -&gt; PARAMETERS [2:27]
+ * |   |   `--PARAMETER_DEF -&gt; PARAMETER_DEF [2:27]
+ * |   |       |--MODIFIERS -&gt; MODIFIERS [2:27]
+ * |   |       |--TYPE -&gt; TYPE [2:27]
+ * |   |       |   `--IDENT -&gt; String [2:27]
+ * |   |       `--IDENT -&gt; name [2:34]
+ * |   |--RPAREN -&gt; ) [2:38]
+ * |   `--SLIST -&gt; { [2:40]
+ * |       |--LITERAL_RETURN -&gt; return [3:8]
+ * |       |   |--EXPR -&gt; EXPR [3:25]
+ * |       |   |   `--PLUS -&gt; + [3:25]
+ * |       |   |       |--STRING_LITERAL -&gt; "Hello, " [3:15]
+ * |       |   |       `--IDENT -&gt; name [3:27]
+ * |       |   `--SEMI -&gt; ; [3:31]
+ * |       `--RCURLY -&gt; } [4:4]
+ * `--RCURLY -&gt; } [5:0]
+ * </pre>
+ * <p><b>-b</b> option shows AST nodes that match given Xpath query. This command can be used to
+ * validate accuracy of Xpath query against given file.</p>
+ * <pre>
+ * $ java -jar checkstyle-X.XX-all.jar Main.java -b "//METHOD_DEF[./IDENT[@text='sayHello']]"
+ * CLASS_DEF -&gt; CLASS_DEF [1:0]
+ * `--OBJBLOCK -&gt; OBJBLOCK [1:18]
+ * |--METHOD_DEF -&gt; METHOD_DEF [2:4]
+ * </pre>
+ * <p>
+ * The following example demonstrates validation of methods order, so that public methods should
+ * come before the private ones:
+ * </p>
+ * <pre>
+ * &lt;module name="MatchXpath"&gt;
+ *  &lt;property name="query" value="//METHOD_DEF[.//LITERAL_PRIVATE and
+ *  following-sibling::METHOD_DEF[.//LITERAL_PUBLIC]]"/&gt;
+ *  &lt;message key=&quot;matchxpath.match&quot;
+ *  value=&quot;Private methods must appear after public methods&quot;/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * public class Test {
+ *  public void method1() { }
+ *  private void method2() { } // violation
+ *  public void method3() { }
+ *  private void method4() { } // violation
+ *  public void method5() { }
+ *  private void method6() { } // ok
+ * }
+ * </pre>
+ * <p>
+ * To violate if there are any parametrized constructors
+ * </p>
+ * <pre>
+ * &lt;module name="MatchXpath"&gt;
+ *  &lt;property name="query" value="//CTOR_DEF[count(./PARAMETERS/*) &gt; 0]"/&gt;
+ *  &lt;message key=&quot;matchxpath.match&quot;
+ *  value=&quot;Parameterized constructors are not allowed&quot;/&gt;
```

Comment:
`Parameterized constructors are not allowed, there should be only default ctor`

this is the only imaginary reason I can think of.

Reply:
done